src/hb-common.cc

```diff
@@ -221,7 +221,13 @@ struct hb_language_item_t {
   }
 
   inline hb_language_item_t & operator = (const char *s) {
-    lang = (hb_language_t) strdup (s);
+    /* If a custom allocated is used calling strdup() pairs
+    badly with a call to the custom free() in finish() below.
+    Therefore don't call strdup(), implement its behavior.
+    */
+    int len = strlen(s) + 1;
```

Comment:
Please, use size_t instead of int, because strlen() returns size_t result type
and memcpy() accepts size with size_t type.